### PR TITLE
Create an SRV record for the etcd instance pool

### PR DESCRIPTION
**Pull request by charlieegan3**
When templating out the terraform config for the etcd stateful instance
pool, this will also create an SRV record in Route53 with a record for
each of the individual A records for each node.

This will eventually allow us to simplify the prometheus scrape config
the 'knowledge' of the number of nodes and their hostnames remains in
terraform only.

dig output:

```
[centos@charlie-cluster|master|ip-10-99-62-26 ~]$ dig etcd-svc.charlie-cluster.tarmak.local. SRV

; <<>> DiG 9.9.4-RedHat-9.9.4-61.el7 <<>> etcd-svc.charlie-cluster.tarmak.local. SRV
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 13292
;; flags: qr rd ra; QUERY: 1, ANSWER: 3, AUTHORITY: 0, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 4096
;; QUESTION SECTION:
;etcd-svc.charlie-cluster.tarmak.local. IN SRV

;; ANSWER SECTION:
etcd-svc.charlie-cluster.tarmak.local. 60 IN SRV 1 10 9115 etcd-2.charlie-cluster.
etcd-svc.charlie-cluster.tarmak.local. 60 IN SRV 1 10 9115 etcd-3.charlie-cluster.
etcd-svc.charlie-cluster.tarmak.local. 60 IN SRV 1 10 9115 etcd-1.charlie-cluster.

;; Query time: 3 msec
;; SERVER: 10.99.0.2#53(10.99.0.2)
;; WHEN: Tue Jun 26 15:32:42 UTC 2018
;; MSG SIZE  rcvd: 192
```

**What this PR does / why we need it**:
This can help make the prometheus scraper config easier (eventually - watch this space)

**Which issue this PR fixes** 
Doesn't  complete it but this is being done as part of https://github.com/jetstack/tarmak/issues/349

**Release note**:

```release-note
Create an SRV record for the etcd instance pool
```

### CODE REVIEW

File: terraform/amazon/templates/instance_pools/instance_pool_instance.tf.template [hunk 1]
```diff
@@ -236,4 +236,16 @@ resource "aws_route53_record" "{{.TFName}}" {
   count   = "${var.{{.TFName}}_min_count}"
 }
 
+{{ if eq .Role.Name "etcd" -}}
+resource "aws_route53_record" "{{.TFName}}-srv-record" {
+  zone_id = "${var.private_zone_id}"
+  name    = "{{.Role.Name}}-svc.${data.template_file.stack_name.rendered}"
```
Comment (simonswine) [inline terraform/amazon/templates/instance_pools/instance_pool_instance.tf.template]: I think maybe we call it `-exporters` instead of `-svc`

Comment (charlieegan3) [reply terraform/amazon/templates/instance_pools/instance_pool_instance.tf.template]: Should be fixed in d356961bd36590a716fc8da9f1bb1124d6387739

**Pull request merged**